lib/apply-extends.js

```diff
@@ -0,0 +1,19 @@
+var fs = require('fs')
+var path = require('path')
+var assign = require('./assign')
+
+function applyExtends (config, cwd) {
+  var defaultConfig = {}
+
+  if (config.hasOwnProperty('extends')) {
+    var pathToDefault = path.join(cwd, config.extends)
```

Comment:
[Comment] I like [@JaKXz's idea](https://github.com/istanbuljs/nyc/issues/503#issuecomment-274916602) of supporting modules by name as well (e.g. `"extends": "my-lib"` instead of just `"extends": "node_modules/my-lib/some-file"`), but since yargs isn't always loading the original "some-file" and doesn't know its path, I like your general approach as the next best thing.